### ISSUE TITLE
fix(model): refuse configured RoPE modes on the theta-only compute HAL

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12608]
# fix(model): refuse configured RoPE modes on the theta-only compute HAL

```routing
lane: model
paths: ["internal/model/qwen35_backend_contract.go", "internal/model/qwen35_backend_contract_test.go"]
expected_steps: 4
```

## Current state

At source snapshot `6872f998b465d96633c3356ef2382c97fc98cc7c`, the model's canonical RoPE path
supports partial rotary widths, layer-specific theta, Llama 3 scaling, YaRN attention factors, and
long-rope factors. The generic compute HAL passes only the full head dimension and one global theta
to `Backend.RoPE`, then admits all non-Qwen3.5 configurations. A two-layer Llama 3 scaled-RoPE
fixture diverges from the legacy CPU reference by `2.5689602e-05`; the identical theta-only control
is bit exact (`max|delta|=0`).

Exact seams at the pinned source are:

- `internal/model/config.go:59-67`, `128-142` defines loaded RoPE scaling and per-layer theta.
- `internal/model/rope.go:14-31`, `170-171` (`invFreq`, `ropeRowForLayer`) consumes every configured axis.
- `internal/compute/compute.go:430` and `internal/compute/cpuref.go:301-311`, `543-563` expose and implement only `(headDim, theta)`.
- `internal/model/hal.go:776-780` (`stepHALMode`) supplies global theta and full head width.
- `internal/model/qwen35_hal.go:667-695` already preserves configured or partial RoPE through a specialized path or exact host fallback, confirming the generic reduction is not the intended semantic contract.
- `internal/model/qwen35_backend_contract.go:162-190` (`ValidateBackendForwardConfig`) nevertheless admits this standard configuration.

Open-issue title/body search on 2026-09-09 found #12534, which optimizes an already specialized
Qwen3.5 partial-RoPE Vulkan shader. It does not cover generic HAL admission or Llama 3 scaling.

## Problem frame

- Centrality: Core
- P1 Context: preserved - backend choice must retain the checkpoint's position encoding over long prompts.
- P2 Net value: advanced - prevents silent position-dependent logit drift across CPU and device HAL sessions.
- P3 Adaptation: N/A - this leaf uses the existing typed architecture refusal until a complete configured-RoPE contract exists.
- P4 Operations: advanced - startup reports the unsupported RoPE mode before weight upload instead of failing only through output quality.

## Why this is next

The generic HAL interface cannot express the valid model configuration already consumed by the
legacy path. A checked constructor that admits it creates plausible but incorrect logits. The
one-file admission predicate restores the documented fail-closed boundary without coupling this
correctness repair to a multi-backend RoPE redesign.

## Parent context

2026-09-09 inference core audit, pinned to `6872f998b465d96633c3356ef2382c97fc98cc7c`.
Coordinate with optimization issue #12534 without treating that device-only work as a fix here.

## Working spine

Loaded configured-RoPE axes -> checked backend admission -> typed unsupported-architecture refusal
-> no generic theta-only RoPE dispatch -> legacy CPU model path remains available.

## Core through-line

Reject non-identity RoPE scaling, partial rotary width, effective long-rope data, and layer-specific
theta at `ValidateBackendForwardConfig` while the generic `Backend.RoPE` contract cannot carry them.
Pin identity/default configs as admitted controls.

## Implementation steps

1. Add a generic-HAL architecture discriminator/error path without changing the dedicated Qwen3.5 hybrid admission contract.
2. Detect each non-identity configured-RoPE axis only when a non-nil backend would enter the generic HAL.
3. Add table tests for axis-specific typed reasons plus nil-backend and theta-only admitted controls.
4. Run the focused checks and the existing default HAL CPU parity witness.

## Gold-plating boundary

- Do not redesign `compute.Backend`, edit GPU shaders, or introduce host readback in this leaf.
- Do not optimize or alter the specialized Qwen3.5 RoPE implementation owned by #12534.
- Do not change loader parsing, cached inverse-frequency math, or the legacy model path.
- Do not weaken or reuse the dedicated Qwen3.5 GDN forward error for this generic-HAL refusal.

## Done condition

- [ ] The checked constructor returns a typed `UnsupportedBackendForwardError` for each configured RoPE mode the generic HAL cannot represent.
- [ ] The error reason identifies the specific non-identity axis: scaling, partial width, long-rope data, or per-layer theta.
- [ ] The ordinary full-head theta-only config and nil-backend legacy path remain admitted.
- [ ] Focused admission tests and `TestHALSessionMatchesLegacyCPUReference` pass.
- [ ] Qwen3.5 hybrid admission retains its current structural capability checks and error identity.

## Definition of done

No configured-RoPE model can silently enter the theta-only generic HAL, while the default Llama
RoPE path remains bit identical and available.

## Witness

This complete before-fix diagnostic includes a matched unscaled control. Save it as
`${TMPDIR:-/tmp}/audit_rope_hal_test.go` and run it through the overlay command below:

```go
package model

import (
	"math"
	"testing"
	"github.com/anthony-chaudhary/fak/internal/compute"
)

func auditRoPEDelta(got, want []float32) (float32, int) {
	var max float32
	at := -1
	for i := range got {
		d := float32(math.Abs(float64(got[i] - want[i])))
		if d > max { max, at = d, i }
	}
	return max, at
}

func TestAuditHALLlama3RoPEParity(t *testing.T) {
	base := Config{
		HiddenSize: 16, NumLayers: 2, NumHeads: 4, NumKVHeads: 2,
		HeadDim: 4, IntermediateSize: 32, VocabSize: 64,
		RMSNormEps: 1e-5, RopeTheta: 500000, TieWordEmbeddings: true,
		EOSTokenID: -1,
	}
	prompt := []int{3, 7, 11, 19, 23, 29}
	control := NewSynthetic(base)
	want0 := control.NewSession().Prefill(prompt)
	hal0, err := control.NewBackendSessionChecked(compute.Pick("cpu-ref"))
	if err != nil { t.Fatalf("unscaled control was refused: %v", err) }
	got0 := hal0.Prefill(prompt)
	d0, at0 := auditRoPEDelta(got0, want0)
	t.Logf("unscaled theta-only control max|delta|=%g at logit %d", d0, at0)
	if d0 != 0 { t.Fatalf("unscaled control drifted: max|delta|=%g", d0) }

	cfg := base
	cfg.RopeScaling, cfg.RopeFactor = "llama3", 8
	cfg.RopeLowFreqFactor, cfg.RopeHighFreqFactor, cfg.RopeOrigContext = 1, 4, 8192
	m := NewSynthetic(cfg)
	want := m.NewSession().Prefill(prompt)
	hal, err := m.NewBackendSessionChecked(compute.Pick("cpu-ref"))
	if err != nil { t.Fatalf("valid llama3 RoPE config was refused: %v", err) }
	got := hal.Prefill(prompt)
	d, at := auditRoPEDelta(got, want)
	if d != 0 {
		t.Fatalf("HAL llama3 RoPE parity: max|delta|=%g at logit %d; got=%g want=%g", d, at, got[at], want[at])
	}
}
```

## Verifiable Witness

From the pinned checkout root:

```bash
repo="$(pwd)"
repro="${TMPDIR:-/tmp}/audit_rope_hal_test.go"
overlay="${TMPDIR:-/tmp}/audit-rope.overlay.json"
printf '{"Replace":{"%s/internal/model/audit_rope_hal_test.go":"%s"}}\n' "$repo" "$repro" > "$overlay"
go test -overlay "$overlay" ./internal/model -run '^TestAuditHALLlama3RoPEParity$' -count=1 -v
```

Observed CPU software failure:

```text
=== RUN   TestAuditHALLlama3RoPEParity
    audit_rope_hal_test.go:29: unscaled theta-only control max|delta|=0 at logit -1
    audit_rope_hal_test.go:48: HAL llama3 RoPE parity: max|delta|=2.5689602e-05 at logit 38; got=-0.12694791 want=-0.12692222
--- FAIL: TestAuditHALLlama3RoPEParity (0.05s)
FAIL
FAIL github.com/anthony-chaudhary/fak/internal/model 0.095s
FAIL
```

The diagnostic above is intentionally a before-fix numerical proof. The landing regression must
assert the concrete typed rejection rather than accepting an arbitrary constructor error.

## Acceptance gate

```text
go test ./internal/model -run '^(TestValidateBackendForwardConfigRejectsConfiguredRoPE|TestHALSessionMatchesLegacyCPUReference)$' -count=1
```

## Likely files

- `internal/model/qwen35_backend_contract.go`
- `internal/model/qwen35_backend_contract_test.go`

## Lane

`model`; two-file admission-correctness leaf, four expected steps.

## Blast radius / fallback

All generic compute backends inherit the theta-only contract. The nil-backend legacy session is the
functional fallback for configured-RoPE models. Default theta-only models are unaffected.
Coordinate edits to `qwen35_backend_contract.go` with the LayerNorm and sliding-window leaves from
the same audit; land them serially if all remain open.

## Closure binding

Close only when the resolving commit cites this issue, carries `(fak model)`, and attaches the
focused green typed-refusal witness. This issue claims CPU software evidence only.
